ethereumj-core/src/test/java/org/ethereum/sharding/ShardingTestHelper.java

```diff
@@ -0,0 +1,192 @@
+import org.ethereum.listener.EthereumListener;
+import org.ethereum.listener.EthereumListenerAdapter;
+import org.ethereum.sharding.config.DepositContractConfig;
+import org.ethereum.sharding.config.ValidatorConfig;
+import org.ethereum.sharding.contract.DepositContract;
+import org.ethereum.sharding.crypto.DepositAuthority;
+import org.ethereum.sharding.crypto.UnsecuredDepositAuthority;
+import org.ethereum.sharding.service.ValidatorRepository;
+import org.ethereum.sharding.service.ValidatorRepositoryImpl;
+import org.ethereum.sharding.service.ValidatorService;
+import org.ethereum.sharding.service.ValidatorServiceImpl;
+import org.ethereum.sharding.util.Randao;
+import org.ethereum.util.blockchain.StandaloneBlockchain;
+import org.ethereum.vm.program.ProgramResult;
+import org.spongycastle.util.encoders.Hex;
+
+import java.math.BigInteger;
+import java.util.Collections;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Future;
+
+/**
+ * @author Mikhail Kalinin
+ * @since 28.07.2018
+ */
+public class ShardingTestHelper {
+
+    public static ShardingBootstrap bootstrap() {
+        ShardingBootstrap sharding = new ShardingBootstrap();
+
+        DepositContractConfig contractConfig = DepositContractConfig.fromFile();
+
+        SystemProperties systemProperties = SystemProperties.getDefault();
+        systemProperties.setGenesisInfo("sharding-test.json");
+        systemProperties.setMineMinGasPrice(BigInteger.ONE);
+        Genesis genesis = Genesis.getInstance(systemProperties);
+
+        StandaloneBlockchain standaloneBlockchain = sharding.standaloneBlockchain = new StandaloneBlockchain()
+                .withGenesis(genesis)
+                .withNetConfig(StandaloneBlockchain.getEasyMiningConfig())
+                .withMinerCoinbase(Hex.decode("cd2a3d9f938e13cd947ec05abc7fe734df8dd826"))
+                .withListener(new CompositeEthereumListener() {
+                    @Override
+                    public void onSyncDone(SyncState state) {
+                        listeners.forEach(listener -> listener.onSyncDone(state));
+                    }
+                });
+        standaloneBlockchain.setSender(
+                ECKey.fromPrivate(Hex.decode("c85ef7d79691fe79573b1a7064c19c1a9819ebdbd1faaab1a8ec92344438aaf4")));
+        standaloneBlockchain.getBlockchain();
+
+        DbFlushManager dbFlushManager = sharding.flushManager = new DbFlushManager(systemProperties, Collections.emptySet(),
+                new WriteCache.BytesKey<>(standaloneBlockchain.getStateDS(), WriteCache.CacheType.SIMPLE)) {
+            @Override
+            public synchronized Future<Boolean> flush() {
+                CompletableFuture<Boolean> ret = new CompletableFuture<>();
+                ret.complete(true);
+                return ret;
+            }
+        };
+        Ethereum ethereum = sharding.ethereum = new EthereumImpl(systemProperties, standaloneBlockchain.getListener()) {
```

Comment:
I don't understand, why it's not executed on StandaloneBlockchain and we need to override it here?

Reply:
`StandaloneBlockchain` doesn't aggregate `Ethereum` facade, actually `StandaloneBlockchain` is a kind of `Ethereum` facade in terms of API it provides. Since entities that are tested here are built on top of `Ethereum` I decided to override its methods rather than introduce an abstraction that allows to use either `Ethereum` or `StandaloneBlockchain` depending on the case

Reply:
So as we could need it further not only for sharding, could you create an issue and link to this override

Reply:
Created a ticket https://github.com/ethereum/ethereumj/issues/1139